sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -270,13 +271,40 @@ private void initBatch(
         vectors[i + partitionIdx].setIsConstant();
       }
     }
+
+    // For Parquet tables whose columns have associated DEFAULT values, this reader must return
+    // those values instead of NULL when the corresponding columns are not present in storage (i.e.
+    // belong to the 'missingColumns' field in this class).
+    ColumnVector[] finalColumns = new ColumnVector[sparkSchema.fields().length];
```

Comment:
This PR targets Orc reader, why needing to change Parquet reader? This looks like newly added code, not moved code.

Reply:
Yes, apologies for confusion, this PR is based off of https://github.com/apache/spark/pull/36672 which changes the Parquet reader. This PR only adds changes for the Orc reader on top of that. Please feel free to wait to review this PR again until the base PR is merged.